client/fingerprint/network_windows.go

```diff
@@ -0,0 +1,47 @@
+package fingerprint
+
+import (
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+// linkSpeed returns link speed in Mb/s, or 0 when unable to determine it.
+func (f *NetworkFingerprint) linkSpeed(device string) int {
+	command := fmt.Sprintf("Get-NetAdapter -IncludeHidden | Where name -eq '%s' | Select -ExpandProperty LinkSpeed", device)
+	path := "powershell.exe"
+	outBytes, err := exec.Command(path, command).Output()
+
+	if err != nil {
+		f.logger.Printf("[WARN] fingerprint.network: Error calling %s (%s): %v", path, command, err)
+		return 0
+	}
+
+	output := strings.TrimSpace(string(outBytes))
```

Comment:
I would pull everything from here down into a parse method and then unit test this method.